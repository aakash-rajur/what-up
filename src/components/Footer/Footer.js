import React from 'react';

function Footer() {
	return (
		<div className="documentation">
			<em>double-click to edit tasks</em><br/>
			Created by <a href="https://github.com/aakashRajur"
			              className="link">aakashRajur</a><br/>
			Inspired by <a href="http://todomvc.com/examples/react/" className="link">React â€¢ TodoMVC</a><br/>
		</div>
	);
}

export default Footer;