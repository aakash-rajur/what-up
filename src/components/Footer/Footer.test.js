import React from 'react';
import Footer from './Footer';

describe('Footer component', () => {
    let component = shallow(<Footer/>);

    it('should render', () => {
        expect(component.find('div.documentation').exists())
            .toBe(true);
    });

    it('dom tree integrity', () => [
            {tag: 'em'}, {tag: 'br'}, {text: 'Created by '},
            {tag: 'a'}, {tag: 'br'}, {text: 'Inspired by '},
            {tag: 'a'}, {tag: 'br'}
        ].forEach(({tag, text}, i) => {
            let child = component.childAt(i);
            expect(tag ? child.type() : child.text())
                .toBe(tag || text)
        })
    );

    it('match snapshot', () => expect(component).toMatchSnapshot());
});
