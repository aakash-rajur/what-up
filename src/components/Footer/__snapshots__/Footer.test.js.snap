// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component match snapshot 1`] = `
<div
  className="documentation"
>
  <em>
    double-click to edit tasks
  </em>
  <br />
  Created by 
  <a
    className="link"
    href="https://github.com/aakashRajur"
  >
    aakashRajur
  </a>
  <br />
  Inspired by 
  <a
    className="link"
    href="http://todomvc.com/examples/react/"
  >
    React â€¢ TodoMVC
  </a>
  <br />
</div>
`;
